code/datums/components/footstep.dm

```diff
@@ -107,20 +99,19 @@
 		return
 	var/mob/living/carbon/human/H = parent
 
+	var/step_type = H.dna.species.footstep_type
+
 	if((H.wear_suit?.body_parts_covered | H.w_uniform?.body_parts_covered | H.shoes?.body_parts_covered) & FEET)
-		// we are wearing shoes
-		playsound(T, pick(GLOB.footstep[T.footstep][1]),
-			GLOB.footstep[T.footstep][2] * volume,
-			TRUE,
-			GLOB.footstep[T.footstep][3] + e_range, falloff_distance = 1, vary = sound_vary)
-	else
-		if(H.dna.species.special_step_sounds)
-			playsound(T, pick(H.dna.species.special_step_sounds), 50, TRUE, falloff_distance = 1, vary = sound_vary)
-		else
-			playsound(T, pick(GLOB.barefootstep[T.barefootstep][1]),
-				GLOB.barefootstep[T.barefootstep][2] * volume,
-				TRUE,
-				GLOB.barefootstep[T.barefootstep][3] + e_range, falloff_distance = 1, vary = sound_vary)
+		step_type = FOOTSTEP_MOB_SHOE // Feet are covered
+
+	var/list/step_sounds = get_step_sounds(step_type)
+	var/turf_footstep = get_turf_sound(T, step_type)
+
```

Comment:
Would perfer this all on one line personally easier to read.

Reply:
Actually compareing it to the other..its akin to the old one..but still hard to read... Effing hell..

Reply:
Don't see how those variables could be combined into a single line, so I'll assume you mean the `playsound()` under them.
For that, yeah... it's pretty bad. But I figured the "lesser of the two evils" there would be to spread out the arguments so that it's a bit clearer to read.